### PR TITLE
Create sequential synthetic data

### LINKED ISSUE #190: Create sequential synthetic data
Add code for creating sequential synthetic data to synth_data_generator.py

### PR DESCRIPTION
**Pull request by frillecode**
Fixes #190

- [ ] I have battle-tested on Overtaci (RMAPPS1279)
- [ ] I have bumped the version as described in the [wiki](https://www.notion.so/Setting-up-Poetry-on-your-local-machine-76647d75e8fa4d9ba553cbc2a49946d9#dca1e0f2ce934a49acd1851aaf882d75)

## Notes for reviewers
Structure is similar to the _src/psycopmlutils/synth_data_generator/synth_data_generator.py_ script, but adapted to produce a column with chunks of generated texts - this is the main part for review.

### CODE REVIEW

File: src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
```
Comment (KennethEnevoldsen) [inline src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py:1]: Add a document docstring stating what the script does (especially when there is a if __name__ ...

File: src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        n_samples (int): Number of rows to generate.
+        df (pd.DataFrame): Dataframe to append to
+        text_prompt (str, optional): Text prompt to use for generating text data. Defaults to "The quick brown fox jumps over the lazy dog".
+
+    Raises:
+        ValueError: If column_type isn't either uniform_int, text datetime_uniform.
+
+    Returns:
+        pd.DataFrame: The generated dataframe.
+    """
```
Comment (KennethEnevoldsen) [inline src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py:67]:
```suggestion
    Returns:
        pd.DataFrame: The generated dataframe.
    
    
    Example:
        >>> column_specifications = {
        >>>   "citizen_ids": {"column_type": "uniform_int", "min": 0, "max": 1_200_000},
        >>>   "timestamp": {"column_type": "datetime_uniform", "min": 0, "max": 5 * 365},
        >>>   "text": {"column_type": "text"},
        >>> }
        >>>
        >>> df = generate_synth_data(
        >>>     predictors=column_specifications,
        >>>     n_samples=100,
        >>>     text_prompt="The patient",
        >>> )
    """
```

Added examples (from button)

**Review by KennethEnevoldsen — COMMENTED** [review]
Few minor changes otherwise looks good.

File: src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py:1]: can be very short

File: src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                df[col] = df_[col]
+
+    return df
+
+
+def generate_data_columns(
+    predictors: Iterable[Dict],
+    n_samples: int,
+    df: pd.DataFrame,
+    text_prompt: str = "The quick brown fox jumps over the lazy dog",
```
Comment (HLasse) [inline src/psycopmlutils/synth_data_generator/synth_txt_data_generator.py:57]:
```suggestion
    text_prompt: str ,
```
More safe to only set defaults in one function and propagate from there

**Review by HLasse — APPROVED** [review]
LGTM!

**Pull request merged**